src/session.ts

```diff
@@ -159,7 +176,7 @@ class NotebookSession {
    *
    * Kill the kernel and shutdown the session.
    */
-  delete(): Promise<void> {
+  dispose(): Promise<void> {
```

Comment:
I think this should probably be `shutdown` just like the kernel. I don't like `dipose` returning a value, because that is contrary to the `IDisposable` interface.